### ISSUE TITLE
DataOut::add_data_vector vector handling

### ISSUE THREAD

**Issue by kronbichler** [dealii/dealii#10474]
The current design of `DataOut::add_data_vector` seems to be a bit annoying for several reasons:
- The vectors passed to `add_data_vector` need to survive until "build_patches" - or is it the write_out functions? This makes writing modular code where the `add_data_vector` function is called in a different module than `build_patches()`/`write_...` cumbersome because one cannot use locally created vectors for output. This hit me more than once, even though I know the documentation.
- One needs to make sure that the vector has ghost values updated, forcing the user to write additional code even though we would know inside of the function what kind of data we need.

It would seem that this is the ideal place to also remove some of the vector templates by either equipping `add_data_vector` to copy itself to a `LinearAlgebra::read_write_vector`. Well, after the discussion at the workshop we might also only allow to pass in that kind of vectors in the first place, but that would not address the first point listed above. What do others think?

@nfehn @peterrum FYI

**Comment by bangerth** [comment]
Yes, I think it would be a nice simplification to just copy the vector. (And to answer your question: The vector only needs to live to `build_patches()` where everything is converted into an internal representation. It doesn't have to live till `write_*()` time.)

**Comment by peterrum** [comment]
I think we can close this issue?

**Comment by kronbichler** [comment]
Let us close this issue, I think the main concept is now implemented with #11594 indeed.
